### ISSUE TITLE
Error detecting BASE_DIR as string

### ISSUE THREAD

**Issue by shacker** [ehmatthes/django-simple-deploy#175]
My first attempt at deploy attempt to fly yielded:

```
... simple_deploy.py", line 375, in _find_git_dir
TypeError: unsupported operand type(s) for /: 'str' and 'str'
```

This was because I had this in settings.py:

`BASE_DIR = os.path.dirname(os.path.dirname(__file__))`

i.e. I was setting BASE_DIR as a string rather than a Path object.  Django handles that fine, but simple-deploy does not. Switching to this fixed it:

`BASE_DIR = Path(__file__).resolve().parent.parent`

The detection code at L375 should either error out gracefully or be modified to handle BASE_DIR as either `str` or `Path` object.

Before completing the test run:
- What OS are you using? MacOS 12.5
- What project are you deploying? (the sample blog project, or your own project?) My own
- If you're not deploying the sample blog project, can you provide a link to the project you're working with?
- https://github.com/shacker/django-todo (but the offending code in my project will be fixed before you see this)
- What dependency management system are you using? venv / requirements.txt / pip-compile
- Which platform are you targeting? fly

**Comment by ehmatthes** [comment]
Thanks, this is really helpful. In Django 3.1, settings.py [started using](https://docs.djangoproject.com/en/4.1/releases/3.1/#miscellaneous) `pathlib.Path` instead of `os.path`. We're not super interested in supporting <3.1. However, there are probably many projects that started pre 3.1, and people updated their projects to newer versions, but did not bother to update their settings file to use `pathlib.Path`. I could see this persisting for quite some time, so simple_deploy should definitely support settings files that use `os.path`.

- [ ] Support settings files that use `os.path` instead of `pathlib.Path`. 
    - [ ] Write a test that fails.
    - [ ] Modify simple_deploy.py so the test passes.
    - [ ] Run integration tests.

**Comment by ehmatthes** [comment]
Write a test that fails

**Comment by ehmatthes** [comment]
This should be fixed in the main branch here, and will be included in the 0.5.8 release.
